shell/platform/android/platform_view_android.cc

```diff
@@ -96,6 +99,19 @@ PlatformViewAndroid::PlatformViewAndroid(
 
 PlatformViewAndroid::~PlatformViewAndroid() = default;
 
+void PlatformViewAndroid::InitSurface() {
```

Comment:
nit: This is kind of a weak abstraction.  It doesn't just init the surface and obfuscates its side effects =)  This would be slightly better:
```c++
surface_factory_ = MakeSurfaceFactory(android_context_);
android_surface_ = MakeSurface(surface_factory_);
```

The benefit being that the side effects are visible at the call site; the dependencies are explicit as input parameters, which also reduces the powers of the abstraction; the abstraction names more closely match what they actually do.

Reply:
Just another point, i've often seen people use `void(void)` methods for initialization and they often end up creating temporal bugs (where one method assumes an instance variable has already been initialized) which can't happen with pure functions.

Reply:
sg, done